src/transformers/modeling_utils.py

```diff
@@ -1432,8 +1432,8 @@ def resize_token_embeddings(
             return model_embeds
 
         # Update base model and current model config
-        self.config.vocab_size = new_num_tokens
-        self.vocab_size = new_num_tokens
+        self.config.vocab_size = model_embeds.weight.shape[0]
```

Comment:
In the line above, if the user doesn't pass a new length in, but DOES pass in a pad_to_multiple_of, the number of embeddings will still change, but this function does an early exit still. I think that this should change to still update the vocab_sizes in the case that the pad_to_multiple_of causes the embeddings to change, even if they didn't pass in a new_num_tokens.

Reply:
This is not the intended use case. We could add this as a feature, but the goal is to pad when we add tokens, not just pad the embedding.

Reply:
Oh - I can see why that would not be the intended use case. I'm not using it that way either. I was just pointing out that this use of the function without that parameter would still cause the model to get into an invalid loadable state.